hack/pre-release.sh

```diff
@@ -148,11 +162,139 @@ update_OCP_version() {
   fi 
 }
 
+# This function handles updating the versions in GitHub
+# takes 3 arguments
+# 1st arg: base branch to operate on 
+# 2nd arg: stream to update. Can be x, y or z to represent the x y or z stream.
+# 3rd arg: current WMCO version
+github_update() {
+  if [ "$#" -ne 3 ]; then
+    echo incorrect parameter count for github_update $#
+    return 1
+  fi
+
+  base_branch=$1
+  stream=$2
+  wmco_version=$3
+
+  # check if the git branch is dirty, and exit if it is
+  check_git_branch
+
+  new_branch=$(create_git_branch_list $stream-stream-bump $base_branch)
+  switch_git_branch $base_branch $new_branch
+
+  update_WMCO_version $stream $wmco_version
+  update_OCP_version $stream $wmco_version
+
+  git submodule update --init --recursive
+  make bundle
+  
+  # remove make bundle artifacts 
+  sed -i 's/REPLACE_IMAGE:latest/REPLACE_IMAGE:/' bundle/manifests/windows-machine-config-operator.clusterserviceversion.yaml
+  sed -i 's/operator-sdk-v1.14.0+git/operator-sdk-v1.15.0+git/' bundle/manifests/windows-machine-config-operator.clusterserviceversion.yaml
+
+  updated_version=$(increment_version $version $stream)
```

Comment:
seems odd you have to do this -- this is the third time you call `increment_version` within this function (it is also called above from update_WMCO_version and update_OCP_version). can update_WMCO_version return the updated version?

Reply:
I agree, but I was thinking that it wouldn't make sense for just update_wmco_version to return the incremented version 
I could pass the incremented version in as an argument, but I felt like that would be kind of odd too.

Reply:
Why does it not make sense for `update_wmco_version` to return the incremented version? Nor do I see passing the version you want to set to `update_wmco_version` as odd. But calling `increment_version` all over the place is definitely inefficient and could lead to bugs. Ideally we should increment the version in one place and then pass it to functions that update files in the repo.